credentialproviderpackage/pkg/configurator/linux/linux.go

```diff
@@ -91,10 +93,26 @@ func (c *linuxOS) UpdateCredentialProvider(_ string) error {
 	}
 
 	out := strings.Join(lines, "\n")
-	err = ioutil.WriteFile(c.extraArgsPath, []byte(out), 0o644)
+	err = os.WriteFile(configPath, []byte(out), 0o644)
```

Comment:
Just curious why are we changing from ioutil to os right now? Are there other references to ioutil in this file?

Reply:
ioutil.readfile was deprecated it basically calls os.readfile, so I updated it, let me know if we should update this with this change. I noticed the same in other files, I can refactor those too.

https://pkg.go.dev/io/ioutil#ReadFile